src/ecs/resources/input.rs

```diff
@@ -333,9 +333,9 @@ impl InputHandler {
     }
 
     /// Returns the value of an axis by the i32 id, if the id doesn't exist this returns None.
-    pub fn axis_value(&self, id: i32) -> Option<f32> {
+    pub fn axis_value(&self, id: &str) -> Option<f32> {
```

Comment:
If we use `Into<String>` we'll likely be introducing an unnecessary allocation, as the hashmap prefers to take keys by reference rather than by value.  Most usage of this is going to be string literals.